core/domain/acl_decorators.py

```diff
@@ -550,3 +551,24 @@ def test_login(self, **kwargs):
     test_login.__wrapped__ = True
 
     return test_login
+
+
+def can_edit_topic(handler):
+    """Decorator to check whether the user can edit given topic."""
+    def test_can_edit(self, topic_id, **kwargs):
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        topic_rights = topic_services.get_topic_rights(topic_id)
+        if topic_rights is None:
+            raise base.UserFacingExceptions.PageNotFoundException
+
+        if topic_services.check_can_manage_topic(
+                self.user, topic_rights):
+            return handler(self, topic_id, **kwargs)
+        else:
+            raise base.UserFacingExceptions.UnauthorizedUserException(
+                'You do not have credentials to edit this exploration.')
```

Comment:
This error message does not make sense in this context.

Reply:
Changed it.